### PR TITLE
Adds Readme to GCP sample app

### PR DESCRIPTION
**Pull request by meltsufin**
Also, fixing a bug in the default constructor of the invoker.

Fixes: #478.

### CODE REVIEW

**Comment by meltsufin** [comment]
Do not merge yet, I'm still having an issue with the sample app when deploying to GCP.
```
java.lang.ClassCastException: class org.eclipse.jetty.server.Request$1 cannot be cast to class [B (org.eclipse.jetty.server.Request$1 is in unnamed module of loader 'app'; [B is in module java.base of loader 'bootstrap') at org.springframework.messaging.converter.StringMessageConverter.convertFromInternal(StringMessageConverter.java:60) at
```
cc/ @dmitry-s @dzou

**Comment by meltsufin** [comment]
The problem above went away.
@olegz  This PR is ready for your review. Thanks!

File: spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml [hunk 1]
```diff
@@ -17,14 +17,15 @@
 	<properties>
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
-		<google.cloud.functions.version>1.0.0-alpha-2-rc3</google.cloud.functions.version>
+		<google.cloud.functions.api.version>1.0.0-alpha-2-rc3</google.cloud.functions.api.version>
```
Comment (dzou) [inline spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml:20]: Should this be rc4?

Comment (meltsufin) [reply spring-cloud-function-adapters/spring-cloud-function-adapter-gcp/pom.xml:20]: rc4 doesn't exist for the API.
See: https://github.com/GoogleCloudPlatform/functions-framework-java/issues/17

**Pull request merged**